File: endpoints/categories.py
from silence.decorators import endpoint

@endpoint(
    route="/categories",
    method="GET",
    sql="SELECT * FROM Categories"
)
def get_all():
    pass
############################

@endpoint(
    route="/categories/$categoryId",
    method="GET",
    sql="SELECT * FROM Categories WHERE categoryId=$categoryId"
)
def get_by_id():
    pass

############################

@endpoint(
    route="/categories/name/$name",
    method="GET",
    sql="SELECT * FROM Categories WHERE name=$name"
)
def get_by_name():
    pass

############################
  
@endpoint(
    route="/categories",
    method="POST",
    sql="INSERT INTO Categories (name, descripcion, foto) VALUES ($name, $descripcion, $foto)",
    description="Creates a new category",
)
def create(name, descripcion, foto):
    pass

###############################################################################

@endpoint(
    route="/categories/$categoryId",
    method="PUT",
    sql="UPDATE Categories SET name = $name, descripcion = $descripcion, foto = $foto WHERE categoryId = $categoryId",
    description="Updates an existing category",
)
def update(name, descripcion, foto):
    pass

###############################################################################

@endpoint(
    route="/categories/$categoryId",
    method="DELETE",
    sql="DELETE FROM Categories WHERE categoryId = $categoryId",
    description="Removes a category",
)
def delete():
    pass
    

File: docs/js/api/categories.js
"use strict";

import { BASE_URL, requestOptions } from "./common.js";

const categoriesAPI = {
    getAll: function () {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/categories`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getById: function (categoryId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/categories/${categoryId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getByName: function (name) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/categories/name/${name}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    create: function (formData) {
        return new Promise(function (resolve, reject) {
            axios
                .post(`${BASE_URL}/categories`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    update: function (categoryId, formData) {
        return new Promise(function (resolve, reject) {
            axios
                .put(`${BASE_URL}/categories/${categoryId}`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    delete: function (categoryId) {
        return new Promise(function (resolve, reject) {
            axios
                .delete(`${ BASE_URL }/categories/${ categoryId }`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });

    }
};
export { categoriesAPI }